plugins/inputs/s7comm/s7comm.go

```diff
@@ -40,7 +40,7 @@ var (
 	}
 	// Word-length mapping taken from https://github.com/robinson/gos7/blob/master/client.go
 	wordLenMap = map[string]int{
-		"X":  0x01, // Bit
+		"X":  0x02, // Bit | Query as byte, as bit address can't be specified in S7DataItem
```

Comment:
Do you still need this?

Reply:
Nope, should be reverted.
2b34129c144ad72b739a6ab3ccd1bcd960527455 is necessary to address @GitTurboy 's Problem.